src/MultichannelMarketing/GLAChannel.php

```diff
@@ -0,0 +1,229 @@
+	}
+
+	/**
+	 * Returns the description of the marketing channel.
+	 *
+	 * @return string
+	 */
+	public function get_description(): string {
+		return __( 'Native integration with Google that allows merchants to easily display their products across Google’s network.', 'google-listings-and-ads' );
+	}
+
+	/**
+	 * Returns the path to the channel icon.
+	 *
+	 * @return string
+	 */
+	public function get_icon_url(): string {
+		return 'https://woocommerce.com/wp-content/uploads/2021/06/woo-GoogleListingsAds-jworee.png';
+	}
+
+	/**
+	 * Returns the setup status of the marketing channel.
+	 *
+	 * @return bool
+	 */
+	public function is_setup_completed(): bool {
+		return $this->merchant_center->is_setup_complete();
+	}
+
+	/**
+	 * Returns the URL to the settings page, or the link to complete the setup/onboarding if the channel has not been set up yet.
+	 *
+	 * @return string
+	 */
+	public function get_setup_url(): string {
+		if ( ! $this->is_setup_completed() ) {
+			return admin_url( 'admin.php?page=wc-admin&path=/google/start' );
+		}
+
+		return admin_url( 'admin.php?page=wc-admin&path=/google/settings' );
+	}
+
+	/**
+	 * Returns the status of the marketing channel's product listings.
+	 *
+	 * @return string
+	 */
+	public function get_product_listings_status(): string {
+		if ( ! $this->merchant_center->is_ready_for_syncing() ) {
+			return self::PRODUCT_LISTINGS_NOT_APPLICABLE;
+		}
+
+		return $this->product_sync_stats->get_count() > 0 ? self::PRODUCT_LISTINGS_SYNC_IN_PROGRESS : self::PRODUCT_LISTINGS_SYNCED;
+	}
+
+	/**
+	 * Returns the number of channel issues/errors (e.g. account-related errors, product synchronization issues, etc.).
+	 *
+	 * @return int The number of issues to resolve, or 0 if there are no issues with the channel.
+	 */
+	public function get_errors_count(): int {
+		try {
+			return $this->merchant_statuses->get_issues()['total'];
+		} catch ( Exception $e ) {
+			return 0;
+		}
+	}
+
+	/**
+	 * Returns an array of marketing campaign types that the channel supports.
+	 *
+	 * @return MarketingCampaignType[] Array of marketing campaign type objects.
+	 */
+	public function get_supported_campaign_types(): array {
+		return $this->campaign_types;
+	}
+
+	/**
+	 * Returns an array of the channel's marketing campaigns.
+	 *
+	 * @return MarketingCampaign[]
+	 */
+	public function get_campaigns(): array {
+		if ( ! $this->ads->ads_id_exists() ) {
+			return [];
+		}
+
+		try {
+			$currency = $this->wc->get_woocommerce_currency();
+
+			return array_map(
+				function ( array $campaign_data ) use ( $currency ) {
+					$cost = null;
+					if ( isset( $campaign_data['amount'] ) ) {
+						$cost = new Price( (string) $campaign_data['amount'], $currency );
+					}
+
+					return new MarketingCampaign(
+						(string) $campaign_data['id'],
+						$this->campaign_types['google-ads'],
+						$campaign_data['name'],
+						admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
```

Comment:
⚠️ This URL doesn't seem right (after testing for woocommerce/woocommerce#36222):

```suggestion
						admin_url( 'admin.php?page=wc-admin&path=/google/dashboard&subpath=/campaigns/edit&programId=' . $campaign_data['id' ),
```